src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -83,7 +83,7 @@ public static int GetGenericDictionarySlot(NodeFactory factory, TypeDesc type)
         /// </summary>
         public static bool HasGenericDictionarySlot(this TypeDesc type)
         {
-            return !type.IsInterface &&
+            return !type.IsInterface && !type.IsCanonicalDefinitionType(CanonicalFormKind.Any) &&
```

Comment:
`type.HasInstantiation` might be a better check to eliminate __Canon/__UniversalCanon.